File: examples/csharp/complex/DecoratedDeepFlowEntry.cs
using System;

namespace Pipeline.Entry
{
    public class DfcWrappedAuditedAttribute : Attribute { }

    public class DecoratedDeepFlowEntry
    {
        [DfcWrappedAudited]
        public void RunDecorated(string rawInput)
        {
            string[] tokens = DfcCommandParser.Tokenize(rawInput);
            string action = DfcCommandParser.ExtractAction(tokens);
            string validated = DfcInputValidator.ValidateLength(action);
            string normalized = DfcInputValidator.NormalizeWhitespace(validated);

            var ctx = new DfcRequestContext(normalized, "cli");
            var registry = new DfcPipelineRegistry();
            registry.Register("default", ctx);
            var registeredCtx = registry.Lookup("default");

            string prefixed = DfcTransformStage.ApplyPrefix(registeredCtx);
            string wrapped = DfcTransformStage.WrapInQuotes(prefixed);

            var assembler = new DfcCommandAssembler();
            assembler.SetPayload(wrapped);
            assembler.SetTarget("cmd.exe");
            string assembled = assembler.Assemble();

            string formatted = DfcOutputFormatter.FormatForShell(assembled);

            var executor = new DfcCommandExecutor();
            string prepared = executor.Prepare(formatted);
            string finalCommand = executor.ConstructCommand(prepared);

            var execCtx = new DfcExecutionContext();
            execCtx.SetCommand(finalCommand);
            string cmdToRun = execCtx.GetCommand();

            var launcher = new DfcProcessLauncher();
            launcher.CreateProcess(cmdToRun);
            launcher.ConfigureEnvironment();
            launcher.Launch();
        }
    }
}

File: examples/kotlin/complex/DecoratedDeepFlowEntry.kt
package com.pipeline.entry

annotation class DfcWrappedAudited

class DecoratedDeepFlowEntry {
    @DfcWrappedAudited
    fun runDecorated(rawInput: String) {
        val tokens = DfcCommandParser.tokenize(rawInput)
        val action = DfcCommandParser.extractAction(tokens)
        val validated = DfcInputValidator.validateLength(action)
        val normalized = DfcInputValidator.normalizeWhitespace(validated)

        val ctx = DfcRequestContext(normalized, "cli")
        val registry = DfcPipelineRegistry()
        registry.register("default", ctx)
        val registeredCtx = registry.lookup("default") ?: return

        val prefixed = DfcTransformStage.applyPrefix(registeredCtx)
        val wrapped = DfcTransformStage.wrapInQuotes(prefixed)

        val assembler = DfcCommandAssembler()
        assembler.setPayload(wrapped)
        assembler.setTarget("/bin/sh")
        val assembled = assembler.assemble()

        val formatted = DfcOutputFormatter.formatForShell(assembled)

        val executor = DfcCommandExecutor()
        val prepared = executor.prepare(formatted)
        val finalCommand = executor.constructCommand(prepared)

        val execCtx = DfcExecutionContext()
        execCtx.setCommand(finalCommand)
        val cmdToRun = execCtx.getCommand()

        DfcAuditLogger.logPreExecution(cmdToRun)

        val launcher = DfcProcessLauncher()
        launcher.createProcess(cmdToRun)
        launcher.configureEnvironment()
        launcher.launch()
    }
}
